### PR TITLE
Do not become catatonic on abort.

### PR DESCRIPTION
**Pull request by mcovarr**

### CODE REVIEW

File: engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/EngineJobExecutionActor.scala [hunk 1]
```diff
@@ -143,8 +133,7 @@ class EngineJobExecutionActor(replyTo: ActorRef,
   }
 
   when(BackendIsCopyingCachedOutputs) {
-    // It's possible that cache writing was turned off due to a `HashError`, hence the `writeToCache` guard.
-    case Event(response: SucceededResponse, data @ ResponsePendingData(_, _, Some(Success(hashes)))) if effectiveCallCachingMode.writeToCache =>
+    case Event(response: SucceededResponse, data @ ResponsePendingData(_, _, Some(Success(hashes)))) =>
```
Comment (cjllanwarne) [inline engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/EngineJobExecutionActor.scala]: Did something change to make the guard no longer needed?

**Comment by cjllanwarne** [comment]
Assuming the deguarding is justified, 👍

File: engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/EngineJobExecutionActor.scala (hunk 1 above)
Comment (mcovarr) [inline engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/EngineJobExecutionActor.scala]: I think if there are hashes in the state data then the hashing must have fully completed successfully and cache writing wouldn't have been turned off.

**Comment by kshakir** [comment]
👍

**Pull request merged**